src/regex.cr

```diff
@@ -275,14 +305,13 @@ class Regex
     String.build do |result|
       str.each_byte do |byte|
         case byte.unsafe_chr
+        # See `Regex::SPECIAL_CHARACTERS`
         when ' ', '.', '\\', '+', '*', '?', '[',
              '^', ']', '$', '(', ')', '{', '}',
              '=', '!', '<', '>', '|', ':', '-'
           result << '\\'
-          result.write_byte byte
-        else
-          result.write_byte byte
         end
+        result.write_byte byte
```

Comment:
Have you benchmarked this change too?

Reply:
Benchmark it.

Reply:
Benchmark what?! Check the diff please!

Reply:
No, they're not:

```
Regex.escape (old)   3.24M (308.28ns) (±13.40%)       fastest
Regex.escape (new)   2.94M (339.82ns) (±14.48%)  1.10× slower
```

Reply:
What? Where does that come from??

Reply:
Comparing the existing and new functions using `Benchmark.ips`

Reply:
Yes, I benchmarked the entire two escape methods before and after your diff. Is that hard to understand?

The sourcecode is different therefore you're feeding different things into the optimizer. The optimizer is heavilly heuristic based, is it that hard to believe that it would generate different code?

Reply:
Well, in this case it **is** a bit hard for me to grasp _that_ difference or where exactly does it comes from (apart that it's related to the optimizer), but benchmarks indeed show ~10% (2-15 in my tests) difference. I'll revert that.